csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -863,7 +863,18 @@ class PredicateChcker : public IterVisitor {
 } // namespace
 
 PredicateElimination::PredicateElimination(Fusion* fusion) {
-  traverseTo(fusion->outputs());
+  // TODO: Common code as in expr_sort.cpp.
+  // Move to a helper function
+  std::vector<Val*> outs_requiring_codegen;
+  outs_requiring_codegen.reserve(fusion->outputs().size());
+  std::copy_if(
+      fusion->outputs().begin(),
+      fusion->outputs().end(),
+      std::back_inserter(outs_requiring_codegen),
+      [&fusion](Val* out) {
+        return (fusion->getOutputAlias(out).type != AllocationType::Evaluate);
+      });
+  traverseTo(outs_requiring_codegen);
```

Comment:
This removes outputs marked for EE from predicate elimination to avoid errors in lowering analysis.